evalml/model_understanding/graphs.py

```diff
@@ -426,24 +426,22 @@ def partial_dependence(pipeline, X, feature, grid_resolution=100):
             over all samples of X and the values used to calculate those predictions.
 
     """
-    if pipeline.model_family == ModelFamily.BASELINE:
-        raise ValueError("Partial dependence plots are not supported for Baseline pipelines")
+    if not isinstance(X, pd.DataFrame):
+        X = pd.DataFrame(X)
     if not pipeline._is_fitted:
         raise ValueError("Pipeline to calculate partial dependence for must be fitted")
-    if pipeline.model_family == ModelFamily.CATBOOST:
-        pipeline.estimator._component_obj._fitted_ = True
-    if pipeline.model_family == ModelFamily.XGBOOST:
-        if isinstance(pipeline, evalml.pipelines.ClassificationPipeline):
-            pipeline.estimator._estimator_type = "classifier"
-            # set arbitrary attribute that ends in underscore to pass scikit-learn check for is_fitted
-            pipeline.estimator.classes_ = pipeline.classes_
-        elif isinstance(pipeline, evalml.pipelines.RegressionPipeline):
-            pipeline.estimator._estimator_type = "regressor"
-            # set arbitrary attribute that ends in underscore to pass scikit-learn check for is_fitted
-            pipeline.estimator.feature_importances_ = pipeline.feature_importance
-        avg_pred, values = sk_partial_dependence(pipeline.estimator, X=X, features=[feature], grid_resolution=grid_resolution)
-    else:
-        avg_pred, values = sk_partial_dependence(pipeline.estimator._component_obj, X=X, features=[feature], grid_resolution=grid_resolution)
+    if pipeline.model_family == ModelFamily.BASELINE:
+        raise ValueError("Partial dependence plots are not supported for Baseline pipelines")
+    if isinstance(pipeline, evalml.pipelines.ClassificationPipeline):
+        pipeline._estimator_type = "classifier"
+    elif isinstance(pipeline, evalml.pipelines.RegressionPipeline):
+        pipeline._estimator_type = "regressor"
+    pipeline.feature_importances_ = pipeline.feature_importance
+    avg_pred, values = sk_partial_dependence(pipeline, X=X, features=[feature], grid_resolution=grid_resolution)
+
+    # Delete scikit-learn attributes that were temporarily set
+    del pipeline._estimator_type
+    del pipeline.feature_importances_
```

Comment:
Very cool!

One note: if `sk_partial_dependence` throws, we'll never reach this code. You could use `finally` though:
```python
try:
    avg_pred, values = sk_partial_dependence(pipeline, X=X, features=[feature], grid_resolution=grid_resolution)
finally:
    del pipeline._estimator_type
    del pipeline.feature_importances_
```

This will allow the exception to still propagate up the stack because we didn't add an `except` clause, but it will guarantee that the code in `finally` runs before the exception continues on!

Not critical, but it would be cool to add this sometime.